walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> collection = new ArrayList<TimeRange>();
+    collection.add(TimeRange.WHOLE_DAY);
+    for (Event event: events) {
+      splitTimeRange(event.getWhen().start(), event.getWhen().end(), 
+        getTimeRangeWithEvent(event, collection, request), collection);
+    }
+    for (Iterator<TimeRange> it = collection.iterator(); it.hasNext();) {
+      TimeRange timeRange = it.next();
+      if (timeRange.duration() < request.getDuration()) {
+        it.remove();
+      }
+    }
+    return collection;
+  }
+  /**
+   * Split a time range into to ranges between which we find an event.
+   *
+   * @param start The start time of the event (the end of the first range).
+   * @param end The end time of the event (the satrt of the second range).
+   * @param timeRange The time range that we want to split.
+   * @param lTimeRanges The list of time ranges in which we look for the time range to be returned.
+   */
+  private void splitTimeRange(int start, int end, TimeRange timeRange, Collection<TimeRange> lTimeRanges) {
+    if (timeRange == null) {
+      return;
+    } 
+    if (timeRange.start() < start) {
+      TimeRange firstTimeRange = TimeRange.fromStartEnd(timeRange.start(), start, false);
+      lTimeRanges.add(firstTimeRange);
+    } 
+    if (timeRange.end() > end) {
+      TimeRange secondTimeRange = TimeRange.fromStartEnd(end, timeRange.end(), false);
+      lTimeRanges.add(secondTimeRange);
+    }
+    lTimeRanges.remove(timeRange);
+  }
+  /**
+   * Return a time range that overlaps with the event in which at least one of the attendees is present.
```

Comment:
How about change the comment to "Iterate through a list of events and return the TimeRange of the first event that overlaps with the request"?